spec/injectable_spec.rb

```diff
@@ -159,6 +159,38 @@ def call
     end
   end
 
+  context 'with dependencies that have a with: keyword option and other with array' do
+    before do
+      DepWithBothArgs = Class.new do
+        attr_reader :somearg, :options
+        def initialize(somearg, **options)
```

Comment:
I think it would fail if the first argument is a hash, as it would get wrongly merged into kwargs. I'll take a look tomorrow.

Reply:
If you can provide a breaking spec with that we can work on the solution together. As David says, bring in as many edge cases as you can think of :D